rclcpp/include/rclcpp/executor.hpp

```diff
@@ -270,7 +270,7 @@ class Executor
                         }));
     }
     // Use the number of subscriptions to allocate memory in the handles
-    size_t number_of_subscriptions = subs.size();
+    unsigned long number_of_subscriptions = subs.size();
```

Comment:
I am agnostic, but using `size_t` sounds reasonable and will likely be more consistent throughout the code.